mouseion/mouseion/tasks.py

```diff
@@ -0,0 +1,196 @@
+            TextColumn("•", style="aw.b"),
+            HumanReadableProgress(),
+            TextColumn("•", style="aw.b"),
+            console=self.console,
+        ) as progress:
+            task_progress_map: Dict[int, int] = {task.id: progress.add_task(
+                f'{task.description}',
+                total=task.size,
+                fields={'type': task.progress_type}
+            ) for task in self.all_tasks}
+            processes = []
+            while self.tasks or processes:
+                # Start ready tasks
+                ready_tasks = [t for t in self.tasks if t.is_ready(self.completed_tasks)]
+                for t in ready_tasks:
+                    process = Process(target=t.process_function, args=(t.size, t.queue))
+                    processes.append((t.id, process, t.queue))
+                    process.start()
+                    self.tasks.remove(t)
+
+                # Avoid busy waiting if no tasks are ready
+                if not ready_tasks:
+                    time.sleep(0.1)
+
+                # Update progress and check for completed tasks
+                for task_id, process, queue in processes:
+                    while not queue.empty():
+                        progress_value = queue.get()
+                        progress.update(task_progress_map[task_id], advance=1)
+                        if progress_value >= self.get_task_size(task_id):  # Assuming task is completed when progress reaches size
+                            self.completed_tasks.add(task_id)
+                            progress.update(task_progress_map[task_id], completed=True)
+
+                # Clean up completed processes
+                processes = [(task_id, p, q) for task_id, p, q in processes if p.is_alive()]
+
+    def get_task_size(self, task_id: int) -> int:
+        for task in self.all_tasks:
+            if task.id == task_id:
+                return task.size
+        return 0
+
+def task_load_jsonl(size: int, queue: Queue) -> float:
+    progress = 0
+    for i in range(size):
+        time.sleep(0.3)
+        progress += 1
+        queue.put(progress)
+    return progress
+
+def task_tokenize(size: int, queue: Queue) -> float:
+    progress = 0
+    for i in range(size):
+        time.sleep(0.3)
+        progress += 1
+        queue.put(progress)
+    return progress
+
+if __name__ == "__main__":
+    tasks = [
+        Task(
+            id=0,
+            description="Load jsonl",
+            size=10,
+            progress_type=ProgressType.MIBI_PER_SECOND,
+            process_function=task_load_jsonl,
+        ),
+        Task(
+            id=1,
+            description="Tokenize",
+            size=20,
+            progress_type=ProgressType.ITERATIONS_PER_SECOND,
+            process_function=task_tokenize,
+            dependencies=0,
+        ),
+    ]
+
+    processor = TaskProcessor(command="RETRO Tokenizer", max_workers=4)
```

Comment:
should max_workers be parameterized rather than static here?

Reply:
TaskProcessor takes the number of workers - As this is just a sample of usage, the end user can write their own TaksProcessor and set the number of workers (so it is parameterized).